pipelined/bevy_pbr2/src/render/mod.rs

```diff
@@ -595,13 +680,27 @@ pub fn queue_meshes(
             let view_matrix = view.transform.compute_matrix();
             let view_row_2 = view_matrix.row(2);
 
-            for (entity, material_handle, mesh_uniform) in standard_material_meshes.iter() {
-                if !render_materials.contains_key(material_handle) {
+            for (entity, material_handle, mesh_handle, mesh_uniform) in
+                standard_material_meshes.iter()
+            {
+                let mut key = PbrPipelineKey::from_msaa_samples(msaa.samples);
+                if let Some(material) = render_materials.get(material_handle) {
+                    if material
+                        .flags
+                        .contains(StandardMaterialFlags::NORMAL_MAP_TEXTURE)
+                    {
+                        key |= PbrPipelineKey::STANDARDMATERIAL_NORMAL_MAP;
+                    }
+                } else {
                     continue;
                 }
+                if let Some(mesh) = render_meshes.get(mesh_handle) {
+                    if mesh.vertex_attributes.contains(Mesh::ATTRIBUTE_TANGENT) {
```

Comment:
Hashing a long-ish string for each mesh entity can't be good for perf. I think we should probably pre-compute this and store it as a `has_tangents: bool` flag on the mesh.

Reply:
I was mostly plumbing it in so that the presence of every vertex attribute could simply be identified, including custom attributes. Good point though. I didn't do any performance test. Changed.